### ISSUE TITLE
Chapter 1 Solution 3 - URLify is incorrect

### ISSUE THREAD

**Issue by bdalal** [careercup/CtCI-6th-Edition-Python#206]
The solution fails for the following test case `lqxwL9 2T5wSg1YKd9Z3iOFL___` (`_` is used to represent space at the end).

The response needs to be `lqxwL9%202T5wSg1YKd9Z3iOFL` but is instead 
```
AssertionError: Lists differ: ['l', 'l', 'q', 'x', 'w', 'L', '9', '%', '2', '0[82 chars] 'L'] != ['l', 'q', 'x', 'w', 'L', '9', '%', '2', '0', '2[77 chars] 'L']
```

Here's a simpler solution using list comprehension:
Assumes that excess spaces are always at the end
```
def urlify(url, length):
    return list(''.join(['%20' if char == ' ' else char for char in string[:length]]))
```
Not sure if such a solution would be acceptable in an interview, though I can't see why not. This is also faster than the original solution.

**Comment by bdalal** [comment]
The fastest would obviously be:
```
def urlify(url):
    return url.strip().replace(" ", "%20")
```
but again, not sure if that is "cheating".

**Comment by brycedrennan** [comment]
Nice catch! I've fixed the issue in #207 

There are two solutions in the code here, one to demonstrate an algorithm and another to demonstrate a pythonic solution.  Your second solution is essentially the same as the existing pythonic solution one.   In an interview situation you'd just want to ask what type of solution the interviewer would prefer.

Your first solution would probably work but it should be returning a list (not a string).
